src/redux/reducers/authReducers.js

```diff
@@ -1,107 +1,109 @@
     case SIGNUP_REQUEST:
       return {
         ...state,
-        isFetching: true
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching
       };
     case SIGNUP_FAILURE:
       return {
         ...state,
-        isFetching: false,
-        lastUpdated: action.receivedAt
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
       };
     case SIGNUP_SUCCESS:
       return {
         ...state,
-        isFetching: false,
-        lastUpdated: action.receivedAt
+        lastUpdated: action.lastUpdated,
+        isAuthenticated: action.isAuthenticated,
+        isFetching: action.isFetching,
       };
     case LOGIN_REQUEST:
       return {
         ...state,
-        isFetching: true
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching
       };
     case LOGIN_FAILURE:
       return {
         ...state,
-        isFetching: false,
-        lastUpdated: action.receivedAt,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
       };
     case LOGIN_SUCCESS:
       return {
         ...state,
-        isFetching: false,
-        lastUpdated: action.receivedAt,
-        isAuthenticated: action.isAuthenticated
+        lastUpdated: action.lastUpdated,
+        isAuthenticated: action.isAuthenticated,
+        isFetching: action.isFetching,
       };
     case LOGOUT_REQUEST:
       return {
         ...state,
-        lastUpdated: action.receivedAt
+        lastUpdated: action.lastUpdated
       };
     case LOGOUT_FAILURE:
       return {
         ...state,
-        lastUpdated: action.receivedAt
+        lastUpdated: action.lastUpdated
       };
     case LOGOUT_SUCCESS:
       return {
         ...state,
-        user: null,
-        lastUpdated: action.receivedAt,
-        isAuthenticated: false
+        lastUpdated: action.lastUpdated,
+        isAuthenticated: action.isAuthenticated,
+        user: action.user,
       };
-    case REQUEST_USER_INFO:
+    case GET_USER_INFO_REQUEST:
       return {
         ...state,
-        isFetching: true,
-        lastUpdated: action.receivedAt,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching
       };
-    case RECEIVE_USER_INFO_FAILURE:
+    case GET_USER_INFO_FAILURE:
       return {
         ...state,
-        isFetching: false,
-        lastUpdated: action.receivedAt,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
       };
-    case RECEIVE_USER_INFO_SUCCESS:
+    case GET_USER_INFO_SUCCESS:
       return {
         ...state,
-        isFetching: false,
+        lastUpdated: action.lastUpdated,
+        isFetching: action.isFetching,
         user: action.user,
-        lastUpdated: action.receivedAt,
       };
     default:
       return state;
   }
 }
 
-const authReducer = combineReducers({
+const rootReducer = combineReducers({
   authenticate
 });
 
-export default authReducer;
+export default rootReducer;
```

Comment:
I think later on, we'll keep the rootReducer in a separate file because it'll be combining all the reducers from many different files (authReducer, tripReducer...). However, for now, this works!

Reply:
Agreed!